os/kernel/task_monitor/task_monitor.c

```diff
@@ -0,0 +1,75 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <sched.h>
+#include <sys/boardctl.h>
+#include <tinyara/sched.h>
+#include "task_monitor_internal.h"
+
+static int monitored_tasks_list[CONFIG_MAX_TASKS];
+
+void task_monitor_update_list(int pid, int status)
+{
+	monitored_tasks_list[PIDHASH(pid)] = status;
+}
+
+/****************************************************************************
+ * Main Function
+ ****************************************************************************/
+int task_monitor(int argc, char *argv[])
+{
+	int pid_idx;
+	struct tcb_s *tcb;
+
+	/* Initialize the monitored tasks list */
+	for (pid_idx = 0; pid_idx < CONFIG_MAX_TASKS; pid_idx++) {
+		monitored_tasks_list[pid_idx] = NOT_MONITORED;
+	}
+
+	while (1) {
```

Comment:
How about consider expandability??
1. Make kernel demon.
2. Attach each monitor to demon.
Then you can maintain only one demon(one thread)

Reply:
Yes. It is a good idea. I will expand it with another commit.